src/main/java/seedu/address/storage/JsonAdaptedTask.java

```diff
@@ -0,0 +1,76 @@
+package seedu.address.storage;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import seedu.address.commons.exceptions.IllegalValueException;
+import seedu.address.model.task.TaskName;
+import seedu.address.model.task.Module;
+import seedu.address.model.task.Deadline;
+import seedu.address.model.task.Task;
+
+import java.time.format.DateTimeFormatter;
+
+public class JsonAdaptedTask {
+
+    public static final String MISSING_FIELD_MESSAGE_FORMAT = "Task's %s field is missing!";
+
+    private static final DateTimeFormatter DEADLINE_FORMATTER = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm");
+
+    public final String taskName;
+    public final String module;
+    public final String deadline;
+
+    /**
+     * Constructs a {@code JsonAdaptedTask} with the given task details.
+     */
+    @JsonCreator
+    public JsonAdaptedTask(@JsonProperty("taskName") String taskName, @JsonProperty("module") String module,
+            @JsonProperty("deadline") String deadline) {
+        this.taskName = taskName;
+        this.module = module;
+        this.deadline = deadline;
+    }
+
+    /**
+     * Converts a given {@code Task} into this class for Jackson use.
+     */
+    public JsonAdaptedTask(Task source) {
+        taskName = source.getName().fullName;
+        module = source.getModule().module;
+        deadline = source.getDeadline().deadline.format(DEADLINE_FORMATTER);
+    }
+
+    /**
+     * Converts this Jackson-friendly adapted task object into the model's {@code Task} object.
+     *
+     * @throws IllegalValueException if there were any data constraints violated in the adapted task.
+     */
+    public Task toModelType() throws IllegalValueException{
+        if (taskName == null) {
+            throw new IllegalValueException(String.format(MISSING_FIELD_MESSAGE_FORMAT, TaskName.class.getSimpleName()));
+        }
+        if (!TaskName.isValidName(taskName)) {
+            throw new IllegalValueException(TaskName.MESSAGE_CONSTRAINTS);
+        }
+        final TaskName modelTaskName = new TaskName(taskName);
+
+        if (module == null) {
+            throw new IllegalValueException(String.format(MISSING_FIELD_MESSAGE_FORMAT, Module.class.getSimpleName()));
+        }
+        if (!Module.isValidModule(module)) {
+            throw new IllegalValueException(Module.MESSAGE_CONSTRAINTS);
+        }
+        final Module modelModule = new Module(module);
+
+        if (deadline == null) {
+            throw new IllegalValueException(String.format(MISSING_FIELD_MESSAGE_FORMAT, Deadline.class.getSimpleName()));
+        }
+        if (!Deadline.isValidDeadline(deadline)) {
+            throw new IllegalValueException(Deadline.MESSAGE_CONSTRAINTS);
+        }
+        final Deadline modelDeadline = new Deadline(deadline);
```

Comment:
The string constructor is expected to handle human input as well. Perhaps `new Deadline(DEADLINE_PARSER.parse(deadline))` would be more reliable here?